migrate/migrate.go

```diff
@@ -231,28 +241,34 @@ func (m *Migrate) replaceRules(p *moduleParts, modsBeingMigrated []string) error
 		return err
 	}
 
-	repoRule := newGoRepoRule(
-		p.module,
-		version,
-		download,
-		name,
-		p.installs(),
-		patches,
-		deps,
-		licences,
-	)
-	if shouldReplaceFirstPartWithRepoRule {
-		idx := ruleIdx(thirdPartyFile, p.parts[0].rule)
-		thirdPartyFile.Stmt[idx] = repoRule.Call
-	} else {
-		part := append(p.parts, p.binaryParts...)[0]
-		if part.pkg == m.thirdPartyFolder {
-			idx := ruleIdx(thirdPartyFile, part.rule)
-			var stmts []build.Expr // Make sure this is a new slice otherwise we'll modify the underlying slice
-			stmts = append(append(stmts, thirdPartyFile.Stmt[:idx]...), repoRule.Call)
-			thirdPartyFile.Stmt = append(stmts, thirdPartyFile.Stmt[idx:]...)
+	if len(licences) == 0 && m.licences != nil {
+		licences, _ = m.licences.Get(p.module, version)
+	}
+
+	if _, ok := m.existingRepoRules[p.module]; !ok {
+		repoRule := newGoRepoRule(
+			p.module,
+			version,
+			download,
+			name,
+			p.installs(),
+			patches,
+			deps,
+			licences,
+		)
+		if shouldReplaceFirstPartWithRepoRule {
+			idx := ruleIdx(thirdPartyFile, p.parts[0].rule)
+			thirdPartyFile.Stmt[idx] = repoRule.Call
 		} else {
-			thirdPartyFile.Stmt = append(thirdPartyFile.Stmt, repoRule.Call)
+			part := append(p.parts, p.binaryParts...)[0]
+			if part.pkg == m.thirdPartyFolder {
+				idx := ruleIdx(thirdPartyFile, part.rule)
+				var stmts []build.Expr // Make sure this is a new slice otherwise we'll modify the underlying slice
+				stmts = append(append(stmts, thirdPartyFile.Stmt[:idx]...), repoRule.Call)
+				thirdPartyFile.Stmt = append(stmts, thirdPartyFile.Stmt[idx:]...)
+			} else {
+				thirdPartyFile.Stmt = append(thirdPartyFile.Stmt, repoRule.Call)
+			}
 		}
```

Comment:
could we change the order here so we can return early? eg move replacePartsWithAliases before this then if repo rule already exists just return? Id rather we didn't have 3 nested if statements